src/sys/bout_types.cxx

```diff
@@ -1,13 +1,22 @@
+#include <bout/assert.hxx>
 #include <bout_types.hxx>
-
 #include <map>
 
+template <typename T>
+const std::string& saveat(const std::map<T, std::string>& mymap, T t) {
+  auto found = mymap.find(t);
```

Comment:
I'm not sure I understand the origin of the name `saveat` -- should this be `safeAt` or something else?

It might help to add an `AUTO_TRACE` to the start of this routine (also to the `*_STRING` methods I guess) so it's clearer which enum type we're looking at.

Reply:
Yes, thanks, I have fixed the name, and added AUTO_TRACE

Reply:
Sorry, perhaps `safeAt` rather than `saveAt` since exception safety seems to be the reason for doing this?

Reply:
Of course :sweat_smile: - thanks :+1: